QualityControl/test/lib/repositories/LayoutRepository.test.js

```diff
@@ -0,0 +1,206 @@
+        });
+      });
+
+      test('should throw an error if name is not provided', () => {
+        const newLayout = { id: '3', owner_id: 'user3' };
+        sinon.assert.notCalled(jsonFileServiceMock.writeToFile);
+        return assert.rejects(
+          layoutRepository.createLayout(newLayout),
+          (err) => err instanceof Error && err.message === 'layout name is mandatory',
+        );
+      });
+
+      test('should throw an error if id already exists', () => {
+        const newLayout = { id: '2', name: 'New Layout', owner_id: 'user3' };
+        sinon.assert.notCalled(jsonFileServiceMock.writeToFile);
+        return assert.rejects(
+          layoutRepository.createLayout(newLayout),
+          (err) => err instanceof Error && err.message === 'layout with this id (2) already exists',
+        );
+      });
+
+      test('should create a new layout successfully', async () => {
+        const newLayout = { id: '3', name: 'New Layout', owner_id: 'user3' };
+        await layoutRepository.createLayout(newLayout);
+
+        assert.equal(jsonFileServiceMock.data.layouts.length, 3);
+        assert.deepEqual(jsonFileServiceMock.data.layouts[2], newLayout);
+        sinon.assert.calledOnce(jsonFileServiceMock.writeToFile);
+      });
+    });
+
+    suite('update layouts', () => {
+      test('should update a single layout by its id', async () => {
+        const newLayout = { id: '1',
+          name: 'Test Layout Updated',
+          owner_id: 'user1',
+          tabs: [
+            {
+              name: 'Tab1',
+              objects: [{ id: '1', name: 'Object1' }],
+            },
+          ] };
+        const idOfLayoutUpdated = await layoutRepository.updateLayout('1', newLayout);
+        equal(idOfLayoutUpdated, '1');
+        deepEqual(jsonFileServiceMock.data.layouts[0], newLayout);
+        sinon.assert.calledOnce(jsonFileServiceMock.writeToFile);
+      });
+    });
+
+    suite('delete layouts', () => {
+      test('should throw an error if the layoutId does not exist', async () => {
+        const nonExistentLayoutId = 'nonExistentId';
+
+        await assert.rejects(
+          layoutRepository.deleteLayout(nonExistentLayoutId),
+          (err) => err instanceof Error &&
+            err.message === `layout (${nonExistentLayoutId}) not found`,
+        );
+
+        strictEqual(jsonFileServiceMock.data.layouts.length, 3);
+        sinon.assert.notCalled(jsonFileServiceMock.writeToFile);
+      });
+
+      test('should delete an existing layout', async () => {
+        const layoutIdToDelete = '3';
+        const deletedLayoutId = await layoutRepository.deleteLayout(layoutIdToDelete);
+
+        assert.strictEqual(deletedLayoutId, layoutIdToDelete);
+        assert.strictEqual(jsonFileServiceMock.data.layouts.length, 2);
+        assert.deepStrictEqual(
+          jsonFileServiceMock.data.layouts,
+          mockedLayouts,
+        );
+        strictEqual(deletedLayoutId, layoutIdToDelete);
```

Comment:
You should try to be consistent in your usage of `assert.something` and `something`, in this case `strictEqual`